packages/nodes-base/nodes/Set/v2/helpers/utils.ts

```diff
@@ -159,13 +159,26 @@ export const validateEntry = (
 	node: INode,
 	itemIndex: number,
 	ignoreErrors = false,
+	nodeVersion?: number,
 ) => {
 	let entryValue = entry[entry.type];
 	const name = entry.name;
 	const entryType = entry.type.replace('Value', '') as FieldType;
 
+	const description = `To fix the error try to change the type for the field "${name}" or activate the option “Ignore Type Conversion Errors” to apply a less strict type validation`;
+
 	if (entryType === 'string') {
-		if (typeof entryValue === 'object') {
+		if (nodeVersion && nodeVersion > 3 && (entryValue === undefined || entryValue === null)) {
+			if (ignoreErrors) {
+				return { name, value: null };
```

Comment:
Should it be passing throught null/undefined like this? Just read the discussion in Linear
```suggestion
				return { name, value: entryValue };
```

Reply:
@elsmr 
`undefined` is not a valid value in `JSON`, we could not display field with `undefined` value in NDV, so that why `null` was chosen as fallback
same as for other types when Ignore Validation `true` and value is `undefined`:
	`const value = validationResult.newValue === undefined ? null : validationResult.newValue;`